### DISCUSSION TITLE
Is it recommended to set time_units

### DISCUSSION THREAD

**Discussion by hyanwong** [tskit-dev/tskit#2084] [Q&A] [answered]
Is it allowed / recommended to set the `time_units` attribute on a tree sequence to e.g. "generations" or "years", given that there are no established constants defining these strings? I'm thinking we should do this for `tsdate` (https://github.com/tskit-dev/tsdate/issues/189)

**Comment by jeromekelleher** [comment, marked as answer]
It's recommended where they are known I think, yes. We don't have any conventions on the values yet, but I guess we stick to lowercase things like ``"generations"`` and ``"years"`` for now we'll be OK.

Think of it as user-communication for now - users would definitely like to know what the time units of the tree sequence are. We don't have to worry too much about conventions to facilitate that.

**Reply by hyanwong** [reply]
Perfect, thanks. In tsdate I think the units must be "generations" by default, because we use Ne to construct the priors, and I guess we assume this is the "unscaled" value in generations.
